### PR TITLE
store/mockstore/mockcopr, store/mockstore/unistore/cophandler: migrate test-infra to testify

### PR DESCRIPTION
**Pull request by unconsolable**
### What problem does this PR solve?

Issue Number: ref #27652

Problem Summary:

### What is changed and how it works?

What's Changed:
- Migrate test predicates to testify
- Add `t.Parallel()` to improve concurrency
- Update dependency `pingcap/bagder`, refactor store clean in `cop_handler_test.go` to eliminate goroutine leak

### Release note 

```release-note
None
```

### CODE REVIEW

File: store/mockstore/mockcopr/cop_handler_dag_test.go [hunk 1]
```diff
@@ -15,91 +15,89 @@
...
 	timeInLoc = time.Date(2018, 8, 15, 12, 0, 0, 0, loc)
 	timeInUTC = time.Date(2018, 8, 15, 12, 0, 0, 0, time.UTC)
-	c.Assert(timeInLoc.Equal(timeInUTC), IsTrue)
+	require.True(t, timeInLoc.Equal(timeInUTC))
 
 	// test the seconds east of UTC is ignored by the function
 	// constructTimeZone().
 	loc, err = constructTimeZone("UTC", 0)
-	c.Assert(err, IsNil)
+	require.Nil(t, err)
```
Comment (tisonkun) [inline store/mockstore/mockcopr/cop_handler_dag_test.go:69]: try replace all with `require.NoError(t, err)`.

File: store/mockstore/mockcopr/main_test.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
+// Copyright 2021 PingCAP, Inc.
```
Comment (tisonkun) [inline store/mockstore/mockcopr/main_test.go:1]: please correct the license header

File: store/mockstore/unistore/cophandler/main_test.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
+// Copyright 2021 PingCAP, Inc.
```
Comment (tisonkun) [inline store/mockstore/unistore/cophandler/main_test.go:1]: please correct the license header

**Review by disksing — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]

File: store/mockstore/mockcopr/main_test.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	"github.com/pingcap/tidb/util/testbridge"
+	"go.uber.org/goleak"
+)
+
+func TestMain(m *testing.M) {
+	testbridge.WorkaroundGoCheckFlags()
+	opts := []goleak.Option{
+		goleak.IgnoreTopFunction("go.etcd.io/etcd/pkg/logutil.(*MergeLogger).outputLoop"),
+		goleak.IgnoreTopFunction("go.opencensus.io/stats/view.(*worker).start"),
+		goleak.IgnoreTopFunction("github.com/pingcap/goleveldb/leveldb.(*DB).mpoolDrain"),
```
Comment (Ranxy) [inline store/mockstore/mockcopr/main_test.go:29]: Maybe we can avoid this process by actively waiting for a second to allow goleveldb to perform a close?

File: store/mockstore/mockcopr/executor_test.go [hunk 5]
```diff
@@ -28,74 +27,75 @@ import (
...
-
 type testExecutorSuite struct {
 	cluster   *testutils.MockCluster
 	store     kv.Storage
 	mvccStore testutils.MVCCStore
 	dom       *domain.Domain
 }
 
-func (s *testExecutorSuite) SetUpSuite(c *C) {
+func SetUpSuite(t *testing.T) (s *testExecutorSuite, clean func()) {
```
Comment (tisonkun) [inline store/mockstore/mockcopr/executor_test.go:44]: I don't think `SetUpSuite` a proper name. Let me try to push a follow-up for this PR.

File: store/mockstore/mockcopr/main_test.go (hunk 4 above)
Comment (tisonkun) [reply store/mockstore/mockcopr/main_test.go]: updated as 988a878

**Review by tisonkun — APPROVED** [review]

**Comment by unconsolable** [comment]
Got it. Other parts I will self check and fix similar issues.

**Pull request merged**